SL/AA.pm

```diff
@@ -967,7 +967,7 @@ sub transactions {
 		 a.ponumber, a.warehouse_id, w.description AS warehouse,
 		 a.description, a.dcn, pm.description AS paymentmethod,
 		 a.datepaid - a.duedate AS paymentdiff,
-		 ad.address1, ad.address2, ad.city, ad.zipcode, ad.country, vc.email
+		 ad.addressline, ad.additional_addressline, ad.place, ad.zip, ad.country, vc.email
```

Comment:
My question is why is_migrated will be false? I mean when we deploy database changes, all databases will be upgraded to the new structure as far as I can understand. Is there any thing else too you need to do to make is_migrated = true?

Reply:
Using is_migrated will make it important to add good number of additional changes for all queries where ever address is selected from database. I am not in favor of too many changes as more changes increase the chance of introducing bugs which take time to discover. So we do that in unavoidable circumstances.

Reply:
The issue with this is, that the old data does not correspond to the new data
So in address1 there can be stuff that should be in addressline (most of the time) or something that should be in additional_addressline or postoffice, its basically unstructured. And if we were to just rename the fields, we still have the same issue of having unstructured data, but now in structured fields.

We want to be able to differentiate between old/unstructured data and new/stuructured data.
The goal regarding the data would be, to slowly migrate old data to new data.
So If a customer had an address like this:
Address1: Postbox 2512
Address2: Bahnhofstrasse 3
City: Zurich
Zipcode: 8001

we don't know that address1 contains the postoffice and address2 the addressline. But once it is edited again, the person that edits it will see the form is wrong and change it.

The form would be with the new fields so:
Addresslinie: Postbox 2512
Additional Addressline: Bahnhofstrasse 3
Postoffice:
City: Zurich
Zipcode: 8001

And the person will be able to change them accordingly:
Addressline: Bahnhofstrasse 3
Additional Addressline:
Postoffice: Postbox 2512
City: Zurich
Zipcode: 8001

And this could then be saved in the new fields in the db and with isMigrated = true